ext/node/lib.rs

```diff
@@ -489,6 +492,250 @@ deno_core::extension!(deno_node,
+      "internal/cli_table.ts",
+      "internal/console/constructor.mjs",
+      "internal/constants.ts",
+      "internal/crypto/_keys.ts",
+      "internal/crypto/_randomBytes.ts",
+      "internal/crypto/_randomFill.ts",
+      "internal/crypto/_randomInt.ts",
+      "internal/crypto/certificate.ts",
+      "internal/crypto/cipher.ts",
+      "internal/crypto/constants.ts",
+      "internal/crypto/diffiehellman.ts",
+      "internal/crypto/hash.ts",
+      "internal/crypto/hkdf.ts",
+      "internal/crypto/keygen.ts",
+      "internal/crypto/keys.ts",
+      "internal/crypto/pbkdf2.ts",
+      "internal/crypto/random.ts",
+      "internal/crypto/scrypt.ts",
+      "internal/crypto/sig.ts",
+      "internal/crypto/util.ts",
+      "internal/crypto/x509.ts",
+      "internal/dgram.ts",
+      "internal/dns/promises.ts",
+      "internal/dns/utils.ts",
+      "internal/dtrace.ts",
+      "internal/error_codes.ts",
+      "internal/errors.ts",
+      "internal/event_target.mjs",
+      "internal/fixed_queue.ts",
+      "internal/fs/streams.mjs",
+      "internal/fs/utils.mjs",
+      "internal/fs/handle.ts",
+      "internal/hide_stack_frames.ts",
+      "internal/http.ts",
+      "internal/idna.ts",
+      "internal/net.ts",
+      "internal/normalize_encoding.mjs",
+      "internal/options.ts",
+      "internal/primordials.mjs",
+      "internal/process/per_thread.mjs",
+      "internal/querystring.ts",
+      "internal/readline/callbacks.mjs",
+      "internal/readline/emitKeypressEvents.mjs",
+      "internal/readline/interface.mjs",
+      "internal/readline/promises.mjs",
+      "internal/readline/symbols.mjs",
+      "internal/readline/utils.mjs",
+      "internal/stream_base_commons.ts",
+      "internal/streams/add-abort-signal.mjs",
+      "internal/streams/buffer_list.mjs",
+      "internal/streams/destroy.mjs",
+      "internal/streams/duplex.mjs",
+      "internal/streams/end-of-stream.mjs",
+      "internal/streams/lazy_transform.mjs",
+      "internal/streams/passthrough.mjs",
+      "internal/streams/readable.mjs",
+      "internal/streams/state.mjs",
+      "internal/streams/transform.mjs",
+      "internal/streams/utils.mjs",
+      "internal/streams/writable.mjs",
+      "internal/test/binding.ts",
+      "internal/timers.mjs",
+      "internal/url.ts",
+      "internal/util.mjs",
+      "internal/util/comparisons.ts",
+      "internal/util/debuglog.ts",
+      "internal/util/inspect.mjs",
+      "internal/util/types.ts",
+      "internal/validators.mjs",
+      "path/_constants.ts",
+      "path/_interface.ts",
+      "path/_util.ts",
+      "path/_posix.ts",
+      "path/_win32.ts",
+      "path/common.ts",
+      "path/mod.ts",
+      "path/separator.ts",
+      "readline/promises.ts",
+      "repl.ts",
+      "wasi.ts",
+    );
+
+    for (specifier, path) in built_in_modules {
+      esm_sources.push(ExtensionFileSource {
+        specifier,
+        code: ExtensionFileSourceCode::LoadedFromFsDuringSnapshot(
```

Comment:
This needs to be `IncludedInBinary` depending on features. I suggest renaming _all_ `deno_node` modules to `node:` specifiers including private ones. Extract the code part of `include_js_files!()` into a new macro like:
```rust
#[cfg(not(feature = "include_js_files_for_snapshotting"))]
#[macro_export]
macro_rules! extension_source_code {
  ($name:ident $( dir $dir:literal, )? $file:literal) => {
    $crate::ExtensionFileSourceCode::IncludedInBinary(
      include_str!(concat!(env!("CARGO_MANIFEST_DIR"), $("/", $dir,)? "/", $file))
    )
  };
}

#[cfg(feature = "include_js_files_for_snapshotting")]
#[macro_export]
macro_rules! extension_source_code {
  ($name:ident $( dir $dir:literal, )? $file:literal) => {
    $crate::ExtensionFileSourceCode::LoadedFromFsDuringSnapshot(
      std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"))$(.join($dir))?.join($file)
    )
  };
}
```
and then define the customizer like:
```rust
customizer = |ext: &mut deno_core::ExtensionBuilder| {
  ext.esm(ExtensionFileSource {
    specifier: `node:_brotli`,
    code: extension_source_code!(dir "polyfills", "_brotli.js")
  });
  ext.esm(ExtensionFileSource {
    specifier: `node:assert`,
    code: extension_source_code!(dir "polyfills", "assert.ts")
  });
  ext.esm(ExtensionFileSource {
    specifier: `node:buffer`,
    code: extension_source_code!(dir "polyfills", "buffer.ts")
  });
  // ...
}
```

Reply:
Thanks, I handled it a bit differently by adding a new `include_js_files_with_specifier!` macro and `esm_with_specifiers` setting in `extension!`. @mmastrac suggests we should change `esm = []` in `extension!` macro to support `"foo.js" as "bla:bla"` but I'm gonna roll with the simpler solution for now.

Reply:
https://github.com/denoland/deno_core/issues/2